VideoLocker/src/main/java/org/edx/mobile/player/PlayerFragment.java

```diff
@@ -350,25 +355,25 @@ public void onResume() {
         }
     }
 
-    public void handleOnResume(){
+    public void handleOnResume() {
         uiHelper.onResume();
-
         setupController();
 
-        if(curMessageTypes.isEmpty()){
+        if (curMessageTypes.isEmpty()) {
             // display progress until playback actually starts
             showProgress();
         }
 
         // start playback after 300 milli seconds, so that it works on HTC One, Nexus5, S4, S5
         // some devices take little time to be ready
-        handler.postDelayed(unfreezeCallback, 300);
+        if (isPrepared) handler.postDelayed(unfreezeCallback, UNFREEZE_DELAY_MS);
     }
 
     @Override
     public void onPause() {
         super.onPause();
         if ( !isInViewPager ) {
+            if (player != null) isPaused = player.isPaused();
```

Comment:
I think you can use freezing instead of pausing to have the player automatically save the playing state.